jcl/source/common/JclSysInfo.pas

```diff
@@ -4255,69 +4319,113 @@ function GetWindows10ReleaseCodeName: String;
       2004:
          Result := '20H1';
       2009:
-         Result := GetWindows10DisplayVersion;
+         Result := GetWindowsDisplayVersion;
     else
-      Result := '';
+      Result := GetWindowsDisplayVersion;
     end;
   end
   else
-    Result := '';
+    Result := GetWindowsDisplayVersion;
 end;
 
-function GetWindows10ReleaseVersion: String;
+function GetWindowsReleaseVersion: String;
 var
   WindowsReleaseId: Integer;
 begin
   if IsWin10 then
   begin
-    WindowsReleaseId := GetWindows10ReleaseId;
+    WindowsReleaseId := GetWindowsReleaseId;
+    if WindowsReleaseId > 0 then
+    begin
+      if WindowsReleaseId < 2009 then
+        Result := LoadResString(@RsOSVersionWin10) + ', version ' + IntToStr(WindowsReleaseId)
+      else
+        Result := LoadResString(@RsOSVersionWin10) + ', version ' + GetWindowsDisplayVersion
+    end
+    else
+      Result := '';
+  end
+  else
+    if IsWinServer then
+  begin
+    WindowsReleaseId := GetWindowsReleaseId;
     if WindowsReleaseId > 0 then
     begin
       if WindowsReleaseId < 2009 then
-        Result := 'Windows 10, version ' + IntToStr(WindowsReleaseId)
+        Result := LoadResString(@RsOSVersionWinServer) + ', version ' + IntToStr(WindowsReleaseId)
       else
-        Result := 'Windows 10, version ' + GetWindows10DisplayVersion
+        Result := LoadResString(@RsOSVersionWinServer) + ', version ' + GetWindowsDisplayVersion
     end
     else
       Result := '';
   end
+  else
+  if IsWin11 then // And higher versions too?
```

Comment:
The style guide mandates that this one should be on the same line as the `else` above --> `else if IsWinServer then`

Reply:
done